### ISSUE TITLE
Android nuget package build broken in runtime.Linux.Microsoft.NETCore.Runtime.CoreCLR.props

### ISSUE THREAD

**Issue by cydhaselton** [dotnet/coreclr#12989]
The Android conditional set up to exclude `libsosplugin.so` and `createdump` from being included in packages is broken; the following code no longer works:

```
<_PlatformDoesNotSupportSosPlugin Condition="'$(_runtimeOSFamily)' == 'android'">true</_PlatformDoesNotSupportSosPlugin>
```
[link](https://github.com/cydhaselton/coreclr/blob/master/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/runtime.Linux.Microsoft.NETCore.Runtime.CoreCLR.props#L11)

Replacing the line with the following conditionals allows the build to finish successfully...
```
<_PlatformDoesNotSupportSosPlugin Condition="'$(_runtimeOSFamily)' == 'android'">true</_PlatformDoesNotSupportSosPlugin>
<_PlatformDoesNotSupportCreatedump Condition="'$(Platform)' == 'arm'">true</_PlatformDoesNotSupportCreatedump>
```
...but may not be accurate (i.e. other arm64 OSs may build those two files.)

NOTE:  Not sure if this is relevant but running the `build.sh` script with the necessary cross variables does not pick up the correct value for `__DistroRid`; pretty sure that `android.21-arm64` was detected successfully after #10286

**Comment by cydhaselton** [comment]
Copying @gkhanna79

**Comment by cydhaselton** [comment]
Polite ping to @gkhanna79 for input

**Comment by gkhanna79** [comment]
@weshaggard @wtgodbe PTAL.

**Comment by gkhanna79** [comment]
CC @janvorli as well.

**Comment by janvorli** [comment]
@cydhaselton I wanted to give it a try on my local Ubuntu 16.10 machine, but I've bumped into serveral issues:
1. Building the cross file system didn't work (sudo ./cross/build-android-rootfs.sh arm64) - I had to update the libandroid-support-dev and libandroid-support library versions from 20 to 21 (20 was not on the termux.net server anymore):
```diff
-wget -nv -nc http://termux.net/dists/stable/main/binary-$__AndroidArch/libandroid-support-dev_20_$__AndroidArch.deb -O $__Android_Cross_Dir/deb/libandroid-support-dev_20_$__AndroidArch.deb
-wget -nv -nc http://termux.net/dists/stable/main/binary-$__AndroidArch/libandroid-support_20_$__AndroidArch.deb -O $__Android_Cross_Dir/deb/libandroid-support_20_$__AndroidArch.deb
+wget -nv -nc http://termux.net/dists/stable/main/binary-$__AndroidArch/libandroid-support-dev_21_$__AndroidArch.deb -O $__Android_Cross_Dir/deb/libandroid-support-dev_21_$__AndroidArch.deb
+wget -nv -nc http://termux.net/dists/stable/main/binary-$__AndroidArch/libandroid-support_21_$__AndroidArch.deb -O $__Android_Cross_Dir/deb/libandroid-support_21_$__AndroidArch.deb
-dpkg -x $__Android_Cross_Dir/deb/libandroid-support-dev_20_$__AndroidArch.deb $__Android_Cross_Dir/tmp/$__AndroidArch/
-dpkg -x $__Android_Cross_Dir/deb/libandroid-support_20_$__AndroidArch.deb $__Android_Cross_Dir/tmp/$__AndroidArch/
+dpkg -x $__Android_Cross_Dir/deb/libandroid-support-dev_21_$__AndroidArch.deb $__Android_Cross_Dir/tmp/$__AndroidArch/
+dpkg -x $__Android_Cross_Dir/deb/libandroid-support_21_$__AndroidArch.deb $__Android_Cross_Dir/tmp/$__AndroidArch/

```
2. After that the cross fs build completed ok, but running the build command the way it has recommended resulted in failure during the native build configuration phase:
```
CMake Error at src/corefx/System.Globalization.Native/CMakeLists.txt:13 (message):
  Cannot find utypes.h, try installing libicu-dev (or the appropriate package
  for your platform)
```
Does this work correctly for you?

**Comment by janvorli** [comment]
@cydhaselton actually, the issue #2 was caused by running the build-android-rootfs.sh under sudo. Now I get another issue: error: 'ucol_setVariableTop' is deprecated. I need to check why the config didn't detect that the `ucol_setMaxVariable` is present and should be used instead of the deprecated function.

**Comment by janvorli** [comment]
@cydhaselton ok, I've fixed that issue by adding libc++_15.2_aarch64.deb package - the libicu depends on it and so the config couldn't find the symbol since it couldn't resolve the libicu dependencies.

**Comment by janvorli** [comment]
I've tried to build everything and libsosplugin.so and createdump were both excluded correctly.

**Comment by cydhaselton** [comment]
Apologies for taking so long to get to this; work at the day job has ramped up considerably since I opened this issue.

As soon as I can...hopefully this week or weekend...I'll update my branch and attempt to re-build

**Comment by RussKeldorph** [comment]
Please re-open if this is still an issue.
